crates/cheatable-starknet/src/cheatcodes/mock_call.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{cheatcodes::EnhancedHintError, CheatedState};
+use starknet_api::core::{ContractAddress, EntryPointSelector};
+use starknet_api::hash::StarkFelt;
+use std::collections::HashMap;
+
+impl CheatedState {
+    pub fn start_mock_call(
```

Comment:
I can imagine such a situation - someone has a contract with only one `get_balance` function and mocks the `set_balance` function which originally it does not have. In the current implementation, it would pass but maybe we should be panicking in such a case.

It can be considered as a bug and as a feature. As a feature because someone could create a `virtual` contract which does not exist without mocking.

Reply:
I think mocking methods that do not exist have some benefit though I agree it kind of depends on what we want `mock_call` to do exactly.

Reply:
imo it should allow mocking whatever user wants because you don't know what it mocks exactly as there may be no contract under mocked address.

Reply:
Ok, so maybe we should include that in the docs?

Reply:
@war-in what exactly do you suggest to include in docs? The fact that it would not panic when the user provides an invalid contract address or fn_name?

Reply:
I would document the fact that users can mock any function at any address even if there are no contract under it

Reply:
Yeah, but we can, for example, `prank` contract_address which doesn't exist, same goes for `roll` and `warp`, and there is no info about it in docs AFAIK. So, I think `mock_call` behavior and docs are consistent with other cheatcodes.

Reply:
We can still consider adding a note about the possibility of applying a cheatcode to a contract/function that doesn't exist

Reply:
Agree with @abulenok here, we should include it everywhere or nowhere

Reply:
Ok, nvm. I just think this is coll stuff and would be nice to have it mentioned somewhere. It's up to you if you want to create some general note about such a possibility.

Reply:
I agree with @war-in here. It would be inconsistent but it is more relevant for mock_call as it does not trigger contact execution. We can mention that mocking does not require contract being deployed.

Reply:
It would be nice to have it in the docs when this PR is merged. Maybe we can write it for `mock_call` and create an issue for the rest?

Reply:
Updated the docs

Reply:
and submitted an issue https://github.com/foundry-rs/starknet-foundry/issues/464